### ISSUE TITLE
Invalid address error when connected with Microsoft DirectAccess

### ISSUE THREAD

**Issue by elad-ep** [aws/aws-iot-device-sdk-cpp-v2#413]
### Describe the bug

Hi, one of our customer reports that they are using Microsoft DirectAccess product. It appears that when a machine is connected using Microsoft DirectAccess, the AWS IoT SDK loses connection to the cloud. according to the customer "the Direct Access VPN creates a Tunnel adapter iphttpsinterface which is purely on IPv6 and also doesn’t have a physical interface". in the SDK log I can see error AWS_IO_SOCKET_INVALID_ADDRESS. can you please inspect the log and let me know your analysis - why the SDK communication is lost? is it possible to support this scenario?

SDK version 1.15.2.
SDK log sent to jkl@amazon.com.

Thanks!

### Expected Behavior

Connection to cloud should persist

### Current Behavior

Connection to cloud is lost

### Reproduction Steps

Connect a machine to Microsoft DirectAccess

### SDK version used

1.15.2

### Environment details (OS name and version, etc.)

Windows 10

**Comment by jmklix** [comment]
Is this happening when you are running the basic-pub-sub?

**Comment by elad-ep** [comment]
I didn't try with the pub-sub - the issue is in the customer's environment. the host name should be the standard iot endpoint address. it is the same whether the machine is connected Microsoft DirectAccess or not. but only when the machine is connected to Microsoft DirectAccess then iot connection is lost... 
Any other interesting thing found in the log?

**Comment by elad-ep** [comment]
Hi, any update please? did you manage to gain some insight from the logs?

**Comment by jmklix** [comment]
No, didn't see anything else in the logs. Can you step through the program to find out what the [host name length](https://github.com/awslabs/aws-c-io/blob/32962ab27a6fd1504d4893cf04c3964af8624ea5/source/channel_bootstrap.c#L831) is when using Microsoft DirectAccess?
